lib/archethic/networking/ip_lookup.ex

```diff
@@ -17,26 +17,29 @@ defmodule Archethic.Networking.IPLookup do
     provider = get_provider()
 
     ip =
-      with {:ok, ip} <- apply(provider, :get_node_ip, []),
-           true <- Networking.valid_ip?(ip) do
-        Logger.info("Node IP discovered by #{provider}")
-        ip
-      else
-        false ->
-          fallback(provider, "NAT : Private IP")
+      case {apply(provider, :get_node_ip, []), should_validate_node_ip?()} do
```

Comment:
Sorry it's hard to read, one propose could be to do like this:
```elixir
def get_node_ip(provider) do
...
  with {:ok, ip} <- apply_provider(provider, :get_node_ip, []),
       :ok <- validate_ip(ip, should_validate_node_ip?()) do
    Logger.info("Node IP discovered by #{provider}")
    ip
  else
    {:error, reason} ->
      fallback(provider, reason)
  end
...
end


defp validate_ip(ip, true) do
   if Networking.valid_ip?(ip) do
      :ok 
   else
      {:error, :invalid_ip}
   end
end

defp validate_ip(ip, false), do: :ok
```

Reply:
agreed